sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/registry/ShardingOrchestrationFacade.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.orchestration.temp.internal.registry;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.shardingsphere.api.config.RuleConfiguration;
+import org.apache.shardingsphere.core.config.DataSourceConfiguration;
+import org.apache.shardingsphere.core.rule.Authentication;
+import org.apache.shardingsphere.orchestration.center.api.ConfigCenter;
+import org.apache.shardingsphere.orchestration.center.api.RegistryCenter;
+import org.apache.shardingsphere.orchestration.center.configuration.InstanceConfiguration;
+import org.apache.shardingsphere.orchestration.center.configuration.OrchestrationConfiguration;
+import org.apache.shardingsphere.orchestration.temp.internal.configcenter.ConfigCenterServiceLoader;
+import org.apache.shardingsphere.orchestration.temp.internal.registry.config.service.ConfigurationService;
+import org.apache.shardingsphere.orchestration.temp.internal.registry.listener.ShardingOrchestrationListenerManager;
+import org.apache.shardingsphere.orchestration.temp.internal.registry.state.service.StateService;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+
+/**
+ * Sharding orchestration facade.
+ *
+ * @author zhangliang
+ * @author caohao
+ * @author panjuan
+ * @author wangguangyuan 
+ */
+@Slf4j
+public final class ShardingOrchestrationFacade implements AutoCloseable {
+    
+    private final RegistryCenter regCenter;
+    
+    private final ConfigCenter configCenter;
+    
+    private final boolean isOverwrite;
+    
+    @Getter
+    private final ConfigurationService configService;
+    
+    private final StateService stateService;
+    
+    private final ShardingOrchestrationListenerManager listenerManager;
+    
+    public ShardingOrchestrationFacade(final OrchestrationConfiguration orchestrationConfig, final Collection<String> shardingSchemaNames) {
+        InstanceConfiguration registryCenterConfiguration = getConfigurationByOrchestrationType(orchestrationConfig.getInstanceConfigurationMap(),
+                "registry_center");
```

Comment:
I suggest `registry_center` can get with `OrchestrationType`, not string directly.

Reply:
OK. I've got it.